### PR TITLE
complete revamp of basic functionality & structure

### PR DESCRIPTION
**Pull request by rprops**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,9 +17,6 @@ before_deploy:
 
 after_success:
   - Rscript -e 'covr::codecov()'
-
```
Comment (FMKerckhof) [inline .travis.yml:20]: I don't think we use any of this functionality either how, hence it is safe to remove

File: DESCRIPTION [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
 Type: Package
-Title: Phenotyping of microbial Raman data
-Version: 0.1.1
+Title: Phenotyping of microbial Raman spectroscopy data
+Version: 0.2
 Authors@R: c(person("Frederiek-Maarten", "Kerckhof", email = "FrederiekMaarten.Kerckhof@UGent.be", role = c("aut", "cre")),
-             person("Benjamin", "Buysschaert", email = "Benjamin.Buysschaert@UGent.be", role = "ctb"),
-             person("Ruben", "Props", email = "Ruben.Props@UGent.be", role = "aut"),
-             person("Cristina", "Garcia-Timermans", email = "Cristina.GarciaTimermans@UGent.be", role = "aut"),
-             person("Dmitry", "Khalenkow", email = "Dmitry.Khalenkow@Ugent.be", role = "ctb"))
```
Comment (FMKerckhof) [inline DESCRIPTION:9]: Dmitry was added here due to collab on the original Garcia Timermans et al. publication, but indeed has not contributed to the package

File: R/hs_PCA.R [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+#' @export
+
+
+hs_PCA <- function(hs.x, ...) {
+  if (is.null(rownames(hs.x@data$spc)))
+    spectra_ids <-
+      hs.x@data$filename
+  else
+    spectra_ids <- rownames(hs.x@data$spc)
+  hs.PCA <- prcomp(x = hs.x@data$spc, ...)
```
Comment (FMKerckhof) [inline R/hs_PCA.R:25]: if feature extraction is the idea, princomp could be desirable, I am not sure here what the aim of the PCA is.

**Review by FMKerckhof — APPROVED** [review]
It may be worthwhile to surpress the figure-html caching from the vingettes (AFAIK Rmd vignettes can be knitted on package install/packaging), for the rest all is looking good.

File: R/hs_PCA.R (hunk 3 above)
Comment (rprops) [reply R/hs_PCA.R:25]: Primarily for dim reduction/visualization purposes - can add princomp if needed in future

**Pull request merged**